SUMMARY element: click() for invisible SUMMARY should toggle DETAILS element

Ahmad-S792 opened the pull request:
#### 19b515f0605328883ede9f5dcae2c07d571dd209
<pre>
SUMMARY element: click() for invisible SUMMARY should toggle DETAILS element

SUMMARY element: click() for invisible SUMMARY should toggle DETAILS element

<a href="https://bugs.webkit.org/show_bug.cgi?id=245937">https://bugs.webkit.org/show_bug.cgi?id=245937</a>

Reviewed by Tim Nguyen.

Merge - <a href="https://chromium.googlesource.com/chromium/src.git/+/c162ebcbfa71bfbdd8e3fdba2137c382f752b4b2">https://chromium.googlesource.com/chromium/src.git/+/c162ebcbfa71bfbdd8e3fdba2137c382f752b4b2</a>

This is to align Webkit behavior with Gecko (Firefox) and Blink (Chrome).

We had unreasonable renderer() existence check for activate event handling. It is not defined by the standard.

* Source/WebCore/html/HTMLSummaryElement.cpp:
(HTMLSummaryElement::supportsFocus): Remove &quot;renderer&quot; condition
(HTMLSummaryElement::defaultEventHandler): Simplified and removed &quot;renderer&quot; condition
* LayoutTests/fast/html/summary-invisible-click.html: Added Test Case
* LayoutTests/fast/html/summary-invisible-click-expected.txt: Added Test Case Expectations
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-summary-element/activation-behavior-expected.txt: Updated Test Expectations

Canonical link: <a href="https://commits.webkit.org/255073@main">https://commits.webkit.org/255073@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b5e554616278d1eaf8a7a2a3e6b31308f215d576

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91216 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/216 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/21809 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/100869 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/160686 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95221 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/229 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29234 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83579 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97352 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96872 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/184 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77957 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27161 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82127 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81770 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70191 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35360 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/15808 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33156 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16858 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3522 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36942 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39716 "Found 1 new test failure: fast/events/blur-remove-parent-crash.html (failure)") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38867 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35952 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Please rebaseline imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-summary-element/activation-behavior.html since this change progressed that test.

Looks good to me otherwise!

Ahmad-S792 committed:
SUMMARY element: click() for invisible SUMMARY should toggle DETAILS element

SUMMARY element: click() for invisible SUMMARY should toggle DETAILS element

https://bugs.webkit.org/show_bug.cgi?id=245937

Reviewed by Tim Nguyen.

Merge - https://chromium.googlesource.com/chromium/src.git/+/c162ebcbfa71bfbdd8e3fdba2137c382f752b4b2

This is to align Webkit behavior with Gecko (Firefox) and Blink (Chrome).

We had unreasonable renderer() existence check for activate event handling. It is not defined by the standard.

* Source/WebCore/html/HTMLSummaryElement.cpp:
(HTMLSummaryElement::supportsFocus): Remove "renderer" condition
(HTMLSummaryElement::defaultEventHandler): Simplified and removed "renderer" condition
* LayoutTests/fast/html/summary-invisible-click.html: Added Test Case
* LayoutTests/fast/html/summary-invisible-click-expected.txt: Added Test Case Expectations
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-summary-element/activation-behavior-expected.txt: Updated Test Expectations

Canonical link: https://commits.webkit.org/255073@main
